File: Servidor/grapeguardapi/src/main/java/com/raquelbytes/grapeguardapi/ApiMap.java
package com.raquelbytes.grapeguardapi;

public class ApiMap {
    public static final String BASE_URL = "http://localhost:8080";
    /**-------------------------USUARIOS--------------------------------*/
    public static final String USUARIO_BASE = "/usuarios";
    public static final String USUARIO_POR_ID = "/usuario/{ID_usuario}";
    public static final String USUARIO_LOGIN = "/login";
    public static final String USUARIO_REGISTER = "/register";
    public static final String USUARIO_MODIFICAR = "/modify/{ID_usuario}";
    public static final String USUARIO_ELIMINAR = "/delete/{ID_usuario}";

    /**-------------------------VINEDOS--------------------------------*/
    public static final String VINEDO_BASE = "/vinedo";
    public static final String VINEDO_ID = "/{id}";
    public static final String VINEDO_USUARIO_ID = "/usuario/{usuarioId}";
    public static final String VINEDO_USUARIO_NUEVO = "/usuario/{usuarioId}/nuevo";
    public static final String VINEDO_MODIFICAR = "/modify/{id}";
    public static final String VINEDO_ELIMINAR = "/delete/{id}";
    public static final String VINEDO_SUMA_HECTAREAS = "/usuario/{usuarioId}/hectareas/suma";

    /**-------------------------TAREA--------------------------------*/
    public static final String TAREA_BASE = "/tarea";
    public static final String TAREA_POR_VINEDO_ID = "/vinedo/{idVinedo}";
    public static final String TAREA_PENDIENTES_POR_VINEDO_ID = "/vinedo/{idVinedo}/pendientes";
    public static final String TAREA_TERMINADAS_POR_VINEDO_ID = "/vinedo/{idVinedo}/terminadas";
    public static final String TAREA_AGREGAR_POR_VINEDO_ID = "/vinedo/{idVinedo}";
    public static final String TAREA_ELIMINAR_POR_ID = "/delete/{id}";

    /**-------------------------COSECHA--------------------------------*/
    public static final String COSECHA_BASE = "/cosecha";
    public static final String COSECHA_POR_VINEDO_ID = "/vinedo/{idVinedo}";
    public static final String COSECHA_AGREGAR_POR_VINEDO_ID = "/vinedo/{idVinedo}/nuevo";
    public static final String COSECHA_ELIMINAR_POR_ID = "/delete/{id}";
}

File: Movil/GrapeGuard/app/src/main/java/com/raquelbytes/grapeguard/Util/ApiMap.kt
package com.raquelbytes.grapeguard.Util


object ApiMap {
    const val BASE_URL = "http://192.168.1.140:8080"

    // Usuarios
    const val USUARIO_BASE = "/usuarios"
    const val USUARIO_POR_ID = "/usuario/{ID_usuario}"
    const val USUARIO_LOGIN = "/login"
    const val USUARIO_REGISTER = "/register"
    const val USUARIO_MODIFICAR = "/modify/{ID_usuario}"

    // Viñedos
    const val VINEDO_ID = "/vinedo/{id}"
    const val VINEDO_USUARIO_ID = "/vinedo/usuario/{ID_usuario}"
    const val VINEDO_USUARIO_NUEVO = "/vinedo/usuario/{ID_usuario}/nuevo"


    // Tareas
    const val TAREA_POR_VINEDO_ID = "/tarea/vinedo/{idVinedo}"
    const val TAREA_PENDIENTES_POR_VINEDO_ID = "/tarea/vinedo/{idVinedo}/pendientes"
    const val TAREA_TERMINADAS_POR_VINEDO_ID = "/tarea/vinedo/{idVinedo}/terminadas"
    const val TAREA_AGREGAR_POR_VINEDO_ID = "/tarea/vinedo/{idVinedo}"
    const val TAREA_ELIMINAR_POR_ID = "/tarea/delete/{idTarea}"

    // Cosechas
    const val COSECHA_POR_VINEDO_ID = "/cosecha/vinedo/{idVinedo}"
    const val COSECHA_AGREGAR_POR_VINEDO_ID = "/cosecha/vinedo/{idVinedo}/nuevo"
    const val COSECHA_ELIMINAR_POR_ID = "/cosecha/delete/{id}"
}
